lib/iris/cube.py

```diff
@@ -3055,7 +2984,32 @@ def _order(array):
 
         return cube_xml_element
 
-    def copy(self, data=None):
+    def replace(self, data, dtype=None, fill_value=None):
+        """
+        Perform an in-place replacement of the cube data.
+
+        Args:
+
+        * data:
+            Replace the data of the cube with the provided data payload.
+
+        Kwargs:
+
+        * dtype:
+            Replacement for the intended dtype of the realised lazy data.
+
+        * fill_value:
+            Replacement for the cube data fill-value.
+
+        .. note::
+            Data replacement alone will clear the intended dtype
+            of the realised lazy data and the fill-value.
+
+        """
+        self._data_manager.replace(data, realised_dtype=dtype)
+        self.fill_value = fill_value
```

Comment:
Do we really need this at all ?
It's potentially a neater way if you need to transfer `fill_value` along with data.
But... 
  * the only internal use at present seems to be in `Aggregator.post_process`
  * various other areas within iris.analysis are all copying + creating cubes rather than using this